src/closures.c

```diff
@@ -207,35 +228,63 @@ ffi_trampoline_table_alloc (void)
   kern_return_t kt;
   uint16_t i;
 
-  /* Allocate two pages -- a config page and a placeholder page */
   config_page = 0x0;
-  kt = vm_allocate (mach_task_self (), &config_page, PAGE_MAX_SIZE * 2,
+  /* The entire allocation is:
+   *    config_page
+   *    trampoline_segment
+   *
+   * trampoline_segment is:
+   *    trampoline dylib mach-o header (if FFI_TRAMPOLINE_WHOLE_DYLIB)
+   *    trampoline page
+   */
+  kt = vm_allocate (mach_task_self (), &config_page, FFI_TRAMPOLINE_ALLOCATION_PAGE_COUNT * PAGE_MAX_SIZE,
 		    VM_FLAGS_ANYWHERE);
```

Comment:
Note that `ffi_trampoline_table_free()` needs to be updated to avoid leaking memory, as it deallocates `PAGE_MAX_SIZE * 2`.

Reply:
@jeremyhu Just took a quick look at `ffi_closure_free()` on macOS 12.5 using [r2](https://rada.re/), and it does indeed have this memory leak:

<img width="337" alt="Screenshot 2022-08-25 at 20 53 46" src="https://user-images.githubusercontent.com/735197/186745954-e3ddeb01-aab0-405d-904b-32036898459f.png">